src/components/scenes/PasswordLoginScene.tsx

```diff
@@ -77,13 +78,21 @@ export const PasswordLoginScene = (props: Props) => {
         })
       } else {
         console.log(error)
-        const errorMessage = error != null ? error.message : undefined
+        const usernameError = asMaybeUsernameError(error)
+        if (usernameError != null) {
+          setUsernameErrorMessage(s.strings.invalid_account)
+          return
+        }
+
+        const passwordError = asMaybePasswordError(error)
+        if (passwordError != null) {
+          setPasswordErrorMessage(s.strings.invalid_password)
+          return
+        }
 
-        // TODO: Need to localize these error messages
-        if (errorMessage === 'Invalid password')
-          setPasswordErrorMessage(errorMessage)
-        if (errorMessage === 'Account does not exist on server')
-          setUsernameErrorMessage(errorMessage)
+        console.error('Unknown login error: ', error)
```

Comment:
`console.error` creates a full-screen red box on React Native. It's better to use `console.warn` or even just `console.log` for things like this, unless the app has truly crashed in a catastrophic way.